src/screens/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useCallback, useRef, useState } from 'react';
+import { View } from 'react-native';
+import PagerView, { PagerViewOnPageScrollEvent } from 'react-native-pager-view';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import { ButtonMode } from 'src/common/enums';
+import { MainButton } from 'src/components';
+import {
+  EducationContent,
+  ExperienceContent,
+  InterestingContent,
+  StepDots,
+  UserContent,
+} from './components';
+import styles from './styles';
+
+type IOnboardingScreenProps = Record<string, never>;
+
+const numberOfDots = 4;
+const lastStepIndex = 3;
+
+const OnboardingScreen: React.FC<IOnboardingScreenProps> = () => {
+  const [currentStep, setCurrentStep] = useState(0);
+  const pagerRef = useRef<PagerView | null>(null);
+
+  const changeCurrentPage = () => {
+    pagerRef.current?.setPage(currentStep + 1);
+  };
+
+  const handlePageScroll = useCallback(
+    (e: PagerViewOnPageScrollEvent) => {
+      setCurrentStep(e.nativeEvent.position);
+    },
+    [setCurrentStep]
```

Comment:
```suggestion
    []
```

It's safe to omit `useState` setters.